### ISSUE TITLE
log: wrong priority value in syslog on Alpine and OpenBSD

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#8269]
Tarantool versions: all present versions are affected (since 1.7.4-460-ga362eff6a, which implements #2466).

A syslog message contains a number that represents facility and severity. The number is called priority (and the severity value also sometimes called priority). The [RFC 5424](https://www.rfc-editor.org/rfc/rfc5424) defines it this way (see section 6):

> ```
> PRI             = "<" PRIVAL ">"
> PRIVAL          = 1*3DIGIT ; range 0 .. 191
> ```
>
> <...>
>
> The Priority value is calculated by first multiplying the Facility number by 8 and then adding the numerical value of the Severity.

Code that implement this logic is the following:

https://github.com/tarantool/tarantool/blob/b6159217153c4312c93eedfd53927f8565232045/src/lib/core/say.c#L1003-L1009

It uses libc's `LOG_MAKEPRI` macro, which is non-standard and historically defined differently:

```c
#define LOG_MAKEPRI(facility, severity) (((facility) << 3) | (severity))
```

```c
#define LOG_MAKEPRI(facility, severity) ((facility) | (severity))
```

Our code assumes that it is implemented as `facility|severity`, but it is not always so. In particular:

* [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/syslog.html) has no `LOG_MAKEPRI` macro, but it defines an argument of the `syslog()` function this way:

  > Values of the priority argument are formed by OR'ing together a severity-level value and an optional facility value.

  It means that `LOG_LOCAL7` and other facility values are already bit-shifted.

  So I assume that `LOG_MAKEPRI` is defined correctly when it has no the bit-shift operation.
* FreeBSD: the macro is [fixed](https://svnweb.freebsd.org/base/head/sys/sys/syslog.h?r1=23252&r2=23251&pathrev=23252) in 1997 (so it is likely [released](https://en.wikipedia.org/wiki/FreeBSD_version_history) in FreeBSD 3).
* OpenBSD: it seems, there is no such macro (according to 643084843ec54037f498c71c2bc4dd323e320dd0). It either was deleted or was not present at all. We define it in our code, but incorrectly. So our behavior on OpenBSD is incorrect.
* Glibc: the macro was [fixed](https://sourceware.org/bugzilla/show_bug.cgi?id=14347) in 2.17 (the [change](https://github.com/gentoo/glibc/commit/c68257729de8a2165e161b187cf866897f71c502) in a git mirror). CentOS 6 has 2.12, CentOS 7 has 2.17. All (or at least most of) the actual Linux distros has correct definition so.
* Musl: the macro is [incorrect](https://git.musl-libc.org/cgit/musl/tree/include/syslog.h#n21) (despite that it was [reported](https://www.openwall.com/lists/musl/2015/10/06/11) in 2015). (I guess it was forgotten, because the project has no bugtracker.) Alpine is the most popular musl based Linux distribution.

To sum up: OpenBSD and Alpine are affected. CentOS 6 as well. Others likely not.

We shouldn't use the `LOG_MAKEPRI` macro.

Reported by @kolsys in PR #8244.
